peerdroid/src/main/java/rdx/works/peerdroid/data/PeerdroidLink.kt

```diff
@@ -20,21 +20,35 @@ import rdx.works.peerdroid.data.webrtc.WebRtcManager
 import rdx.works.peerdroid.data.webrtc.model.PeerConnectionEvent
 import rdx.works.peerdroid.data.webrtc.model.SessionDescriptionWrapper
 import rdx.works.peerdroid.data.webrtc.model.completeWhenDisconnected
+import rdx.works.peerdroid.data.webrtc.wrappers.datachannel.DataChannelWrapper
 import rdx.works.peerdroid.data.websocket.WebSocketClient
 import rdx.works.peerdroid.data.websocket.model.RpcMessage.AnswerPayload.Companion.toAnswerPayload
 import rdx.works.peerdroid.data.websocket.model.SignalingServerMessage
 import rdx.works.peerdroid.di.ApplicationScope
 import rdx.works.peerdroid.di.IoDispatcher
+import rdx.works.peerdroid.domain.ConnectionIdHolder
 import timber.log.Timber
-import java.lang.IllegalStateException
 
 interface PeerdroidLink {
 
     /**
      * Call this function to add a connection in the wallet settings.
      *
      */
-    suspend fun addConnection(encryptionKey: ByteArray): Result<Unit>
+    suspend fun addConnection(
+        encryptionKey: ByteArray,
+        connectionListener: ConnectionListener
+    ): Result<Unit>
+
+    suspend fun send(
+        message: String,
+        connectionId: String
+    ): Result<Unit>
```

Comment:
this is the only thing that I wouldn't like it to have it here but I completely understand why it had to be implemented this way...

However one proposal for a slightly better improvement it would be to reduce its "scope", because now a developer could (theoretically) think this function can send any message to a CE.

So what if would move the `ConnectorExtensionExchangeInteraction` data model to `peerdroid` module (which it seems to be a part of it) and refactor this fun as:

``` 
 suspend fun sendExchangeInteraction(
        message: ConnectorExtensionExchangeInteraction,
        connectionId: String
    ): Result<Unit>
```

and then you json encode the message in `PeerdroidLink`

Reply:
Good point!